ts/components/conversation/SessionConversation.tsx

```diff
@@ -353,6 +353,15 @@ export class SessionConversation extends React.Component<Props, State> {
             window.inboxStore?.dispatch(resetSelectedMessageIds());
           }
           break;
+        case 'Backspace':
+        case 'Delete':
+          if (selectionMode) {
+            void deleteMessagesByIdForEveryone(
```

Comment:
You need to account for messages that you can only delete locally. For example being a non-admin in a group or a community.

```tsx
            if (selectedConversationKey) {
              if (isOnlyServerDeletable) {
                void onDeleteSelectedMessagesForEveryone(
                  selectedConversationKey,
                  selectedMessageIds
                );
              } else {
                void deleteMessagesById(selectedMessageIds, selectedConversationKey);
              }
            }
  ```

Reply:
const isOnlyServerDeletable = isPublic;